pkg/sm/sm.go

```diff
@@ -165,7 +166,9 @@ func New(ctx context.Context, cancel context.CancelFunc, e env.Environment, cfg
 	}
 
 	operationMaintainer := operations.NewMaintainer(ctx, interceptableRepository, postgresLockerCreatorFunc, cfg.Operations, waitGroup)
-	osbClientProvider := osb.NewBrokerClientProvider(cfg.HTTPClient.SkipSSLValidation, int(cfg.HTTPClient.ResponseHeaderTimeout.Seconds()))
+	osbClientTimeout := math.Min(float64(cfg.HTTPClient.Timeout), float64(cfg.Server.RequestTimeout))
```

Comment:
you subtract a second in the timeout handler so it can write before the deadline, so if server timeout is picked here the timeout handler will timeout one second before the request is interrupted. if the request finishes within this second, probably a 2nd write attempt might be done.